src/libraries/System.IO.MemoryMappedFiles/src/Microsoft/Win32/SafeMemoryMappedFileHandle.Unix.cs

```diff
@@ -59,9 +57,22 @@ public sealed partial class SafeMemoryMappedFileHandle : SafeHandleZeroOrMinusOn
             _options = options;
             _capacity = capacity;
 
-            // Fake a unique int handle value > 0.
-            int nextHandleValue = (int)((Interlocked.Increment(ref s_counter) % (int.MaxValue - 1)) + 1);
-            SetHandle(new IntPtr(nextHandleValue));
+            IntPtr handlePtr;
+
+            if (fileStream != null) {
+                _fileStreamHandle = fileStream.SafeFileHandle;
```

Comment:
This should really be stored only after calling DangerousAddRef.  The issue is that if DangerousAddRef throws, the finalizer for this object will still be invoked, and if it sees _fileStreamHandle as non-null, it'll call DangerousRelease.  Now, I see we do this same ordering in pipes, and in reality it's unlikely to be an actual issue, since the only reason DangerousAddRef should throw is if the handle has already been disposed, in which case an extraneous DangerousRelease on it in the finalizer shouldn't matter functionally.  But, still good hygiene.